trollflow2/tests/test_launcher.py

```diff
@@ -180,117 +178,116 @@ def test_message_to_jobs_minimal(self):
                                                      'productname': 'overview'}}})])
         self.assertDictEqual(jobs[999]['product_list']['product_list'], expected)
 
-class TestRun(unittest.TestCase):
+class TestRun(TestCase):
 
     def setUp(self):
+        super().setUp()
         self.config = yaml.load(yaml_test1, Loader=UnsafeLoader)
 
-    @mock.patch('trollflow2.launcher.yaml.load')
-    @mock.patch('trollflow2.launcher.open')
-    @mock.patch('trollflow2.launcher.process')
-    @mock.patch('trollflow2.launcher.Process')
-    @mock.patch('trollflow2.launcher.ListenerContainer')
-    def test_run(self, lc_, Process, process, _open, yaml_load):
+    def test_run(self):
         from trollflow2.launcher import run
-        listener = mock.MagicMock()
-        listener.output_queue.get.return_value = 'foo'
-        lc_.return_value = listener
-        proc_ret = mock.MagicMock()
-        Process.return_value = proc_ret
-        # stop looping
-        proc_ret.join.side_effect = KeyboardInterrupt
-        yaml_load.return_value = self.config
-        prod_list = 'bar'
-        try:
-            run(prod_list)
-        except KeyboardInterrupt:
-            pass
-        listener.output_queue.called_once()
-        Process.assert_called_with(args=('foo', prod_list), target=process)
-        proc_ret.start.assert_called_once()
-        proc_ret.join.assert_called_once()
-        lc_.assert_called_with(topics=['/topic1', '/topic2'])
-        # Subscriber topics are removed from config
-        self.assertTrue('subscribe_topics' not in self.config['common'])
-        # Topics are given as command line option
-        lc_.reset_mock()
-        try:
-            run(prod_list, topics=['/topic3'])
-        except KeyboardInterrupt:
-            pass
-        lc_.assert_called_with(topics=['/topic3'])
-
-    @mock.patch('trollflow2.launcher.yaml.load')
-    @mock.patch('trollflow2.launcher.open')
-    @mock.patch('trollflow2.launcher.ListenerContainer')
-    def test_run_keyboard_interrupt(self, lc_, _open, yaml_load):
+        with mock.patch('trollflow2.launcher.yaml.load') as yaml_load,\
```

Comment:
Are these context manager versions somehow better than having the decorators? Or is it required as part of the new `TestCase` wrapper class? I think it just adds one unnecessary level of intendation for a long piece of code and creates unnecessarily long diffs.

Reply:
yes, it's not the nicest, but since we import plugins in the function, we need to patch the function after the import

Reply:
and yes, also the patcher needs to be in place, so the patches need to be done inside the test function